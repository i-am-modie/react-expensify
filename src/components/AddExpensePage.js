import React from 'react';

const AddExpensePage = ()=>(
    <p>add</p>
);

export default AddExpensePage;
