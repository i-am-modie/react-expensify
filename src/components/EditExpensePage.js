import React from 'react';

const EditExpensePage = (props)=>(
    <p>editing expense with id of {props.match.params.id}</p>
);

export default EditExpensePage;