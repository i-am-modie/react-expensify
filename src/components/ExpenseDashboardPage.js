import React from 'react';

const ExpenseDashboardPage = ()=>(
    <p>test</p>
);

export default ExpenseDashboardPage;