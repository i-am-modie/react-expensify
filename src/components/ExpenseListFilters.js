import React from "react";
import { connect } from "react-redux";
import {
    setEndDate,
    setStartDate,
    setTextFilter,
    sortAsc,
    sortByAmount,
    sortByDate,
    sortDesc
} from "../actions/filters";
import {DateRangePicker} from "react-dates";

export class ExpenseListFilters extends React.Component {
    state = {
        calendarFocused: null
    };
    onDatesChange = ({startDate, endDate}) => {
        this.props.setStartDate(startDate);
        this.props.setEndDate(endDate);
    };
    onFocusChange = calendarFocused => {
        this.setState(() => ({calendarFocused}));
    };
    onTextChange = e => {
        this.props.setTextFilter(e.target.value);
    };
    onSortChange = e => {
        if (e.target.value === "amount") {
            this.props.sortByAmount();
        } else if (e.target.value === "date") {
            this.props.sortByDate();
        }
    };
    onOrderChange = e => {
        if (e.target.value === "asc") {
            this.props.sortAsc();
        } else if (e.target.value === "desc") {
            this.props.sortDesc();
        }
    };
    render() {
        return (
            <div>
                <input
                    type="text"
                    value={this.props.filters.text}
                    onChange={this.onTextChange}
                />

                <select
                    value={this.props.filters.sortBy}
                    onChange={this.onSortChange}
                >
                    <option value="date">Date</option>
                    <option value="amount">Amount</option>
                </select>

                <select
                    value={this.props.filters.sortOrder}
                    onChange={this.onOrderChange}
                >
                    <option value="asc">^</option>
                    <option value="desc">v</option>
                </select>
                <DateRangePicker
                    startDate={this.props.filters.startDate}
                    startDateId="start"
                    endDate={this.props.filters.endDate}
                    endDateId="end"
                    onDatesChange={this.onDatesChange}
                    focusedInput={this.state.calendarFocused}
                    onFocusChange={this.onFocusChange}
                    showClearDates={true}
                    numberOfMonths={1}
                    isOutsideRange={() => false}
                />
            </div>
        );
    }
}

const mapStateToProps = state => ({
    filters: state.filters
});

const mapDispatchtoProps = dispatch => ({
    setTextFilter: (text) => dispatch(setTextFilter(text)),
    sortByDate: () => dispatch(sortByDate()),
    sortByAmount: () => dispatch(sortByAmount()),
    setStartDate: (startDate) => dispatch(setStartDate(startDate)),
    setEndDate: (endDate) => dispatch(setEndDate(endDate)),
    sortAsc: () => dispatch(sortAsc()),
    sortDesc: () => dispatch(sortDesc()),

});
export default connect(mapStateToProps, mapDispatchtoProps)(ExpenseListFilters);
