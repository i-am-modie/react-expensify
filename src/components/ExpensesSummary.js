import React from "react";
import {connect} from "react-redux";
import {Link} from 'react-router-dom';
import selectExpenses from "../selectors/expenses";
import selectExpensesTotal from "../selectors/expenses-total";
import numeral from "numeral";

export const ExpensesSummary = props => {
    const expenses = props.expenses || [];
    const expenseWord = expenses.length === 1 ? "expense" : "expenses";
    const formattedExpensesTotal = numeral(
        selectExpensesTotal(expenses) / 100
    ).format("0,0[.]00$");
    return (
        <div className="page-header">
            <div className="content-container">
                <h1 className="page-header__title">
                    Viewing <span>{expenses.length}</span> {expenseWord} totalling{" "}
                    <span>{formattedExpensesTotal}</span>
                </h1>
                <div className="page-header__actions">
                    <Link className="button" to="/create">Add Expense</Link>
                </div>
            </div>
        </div>
    );
};

const mapStateToProps = state => ({
    expenses: selectExpenses(state.expenses, state.filters)
});

export default connect(mapStateToProps)(ExpensesSummary);
