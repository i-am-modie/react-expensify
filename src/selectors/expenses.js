export default (expenses, { text, sortBy, sortOrder, startDate, endDate }) => {
    let order = sortOrder === 'desc' ? 1 : -1;
    return expenses
        .filter(expense => {
            const startDateMatch =
                typeof startDate !== "number" || expense.createdAt >= startDate;
            const endDateMatch =
                typeof endDate !== "number" || expense.createdAt <= endDate;
            const textMatch = expense.description.match(new RegExp(text, "i"));
            return startDateMatch && endDateMatch && textMatch;
        })
        .sort((a, b) => {
            if (sortBy === "date") {
                return a.createdAt < b.createdAt ? order : -order;
            }
            if (sortBy === 'amount') {
                return a.amount < b.amount ? order : -order;
            }
        });
};