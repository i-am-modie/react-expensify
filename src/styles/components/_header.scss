.header {
  background-color: $gray;
}

.header__title {
  color: white;
  text-decoration: none;
  h1 {
    margin: 0;
  }
}

.header__content {
  align-items: center;
  display: flex;
  justify-content: space-between;
  padding: $s-size 0;
}