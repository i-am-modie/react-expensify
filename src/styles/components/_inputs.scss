.text-input {
  border: 1px solid #cacccd;
  height: 50px;
  font-style: $font-size-large;
  font-weight: 300;
  padding: $s-size;
  @media (max-width: $desktop-breakpoint) {
    width: 100%;
  }
}

.DateRangePicker, .DateRangePickerInput {
  @media (max-width: $desktop-breakpoint) {
    width: 100%;
  }
}

.select {
  @extend .text-input;
}

.textarea {
  @extend .text-input;
  height: 10rem;
}