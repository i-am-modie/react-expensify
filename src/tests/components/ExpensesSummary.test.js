import {ExpensesSummary} from '../../components/ExpensesSummary';
import expenses from '../fixtures/expenses';
import React from 'react';
import {shallow} from 'enzyme';


test('should correctly render ExpensesSummary with no expenses', () => {
    const wrapper = shallow(<ExpensesSummary/>);
    expect(wrapper).toMatchSnapshot();
});

test('should correctly render ExpensesSummary with one expense', () => {
    const wrapper = shallow(<ExpensesSummary expenses={[expenses[0]]}/>);
    expect(wrapper).toMatchSnapshot();
});
test('should correctly render ExpensesSummary with multiple expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenses={expenses}/>);
    expect(wrapper).toMatchSnapshot();
});