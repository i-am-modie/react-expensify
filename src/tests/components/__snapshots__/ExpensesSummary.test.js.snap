// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render ExpensesSummary with multiple expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        3
      </span>
       
      expenses
       totalling
       
      <span>
        1,996.95$
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`should correctly render ExpensesSummary with no expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        0
      </span>
       
      expenses
       totalling
       
      <span>
        0$
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`should correctly render ExpensesSummary with one expense 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        1
      </span>
       
      expense
       totalling
       
      <span>
        1.95$
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
